hikari/events/message_events.py

```diff
@@ -214,6 +214,41 @@ def author(self) -> users.User:
             The user that sent the message.
         """
 
+    @property
+    def is_bot(self) -> bool:
+        """Return `builtins.True` if the message is from a bot.
+
+        Returns
+        -------
+        builtins.bool
+            `builtins.True` if from a bot, or `builtins.False` otherwise.
+        """
+        return self.message.author.is_bot
+
+    @property
+    def is_webhook(self) -> bool:
+        """Return `builtins.True` if the message was created by a webhook.
+
+        Returns
+        -------
+        builtins.bool
+            `builtins.True` if from a webhook, or `builtins.False` otherwise.
+        """
+        return self.message.webhook_id is not None
+
+    @property
+    def is_human(self) -> bool:
+        """Return `builtins.True` if the message was created by a human.
+
+        Returns
+        -------
+        builtins.bool
+            `builtins.True` if from a human user, or `builtins.False` otherwise.
+        """
+        # Not second-guessing some weird edge case will occur in the future with this,
+        # so I am being safe rather than sorry.
+        return not self.message.author.is_bot and self.message.webhook_id is None
```

Comment:
If we're account for edge cases should we not consider checking for if it's a system user or are the definitions between what might account for a system users in the future a bit too blured for that to be accurate. (not too sure about this but what i do know is discord isn't consistent with setting the system user flag as per https://github.com/discord/discord-api-docs/issues/1994)

Reply:
Hmm, isn't the system user still a bot/webhook?

Reply:
It should be but I feel like we could only be as sure of that as we could be sure that a webhook message would also be created by a bot. But this shouldn't matter too much, would be too hard to second guess what discord's going to do with system users anyway

Reply:
We can resolve this another time when clarification is clearer I think.